internal/manifests/mutate.go

```diff
@@ -0,0 +1,349 @@
+
+		default:
+			t := reflect.TypeOf(existing).String()
+			return fmt.Errorf("missing mutate implementation for resource type: %s", t)
+		}
+		return nil
+	}
+}
+
+func mergeWithOverride(dst, src interface{}) error {
+	return mergo.Merge(dst, src, mergo.WithOverride)
+}
+
+func mutateSecret(existing, desired *corev1.Secret) {
+	existing.Labels = desired.Labels
+	existing.Annotations = desired.Annotations
+	existing.Data = desired.Data
+}
+
+func mutateConfigMap(existing, desired *corev1.ConfigMap) {
+	existing.BinaryData = desired.BinaryData
+	existing.Data = desired.Data
+}
+
+func mutateServiceAccount(existing, desired *corev1.ServiceAccount) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+}
+
+func mutateClusterRole(existing, desired *rbacv1.ClusterRole) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Rules = desired.Rules
+}
+
+func mutateClusterRoleBinding(existing, desired *rbacv1.ClusterRoleBinding) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Subjects = desired.Subjects
+}
+
+func mutateRole(existing, desired *rbacv1.Role) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Rules = desired.Rules
+}
+
+func mutateRoleBinding(existing, desired *rbacv1.RoleBinding) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Subjects = desired.Subjects
+}
+
+func mutateV2BetaHPA(existing, desired *autoscalingv2beta2.HorizontalPodAutoscaler) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Spec = desired.Spec
+}
+
+func mutateAutoscalingHPA(existing, desired *autoscalingv2.HorizontalPodAutoscaler) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Spec = desired.Spec
+}
+
+func mutateIngress(existing, desired *networkingv1.Ingress) {
+	existing.Labels = desired.Labels
+	existing.Annotations = desired.Annotations
+	existing.Spec.DefaultBackend = desired.Spec.DefaultBackend
+	existing.Spec.Rules = desired.Spec.Rules
+	existing.Spec.TLS = desired.Spec.TLS
+}
+
+func mutateRoute(existing, desired *routev1.Route) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Spec = desired.Spec
+}
+
+func mutateServiceMonitor(existing, desired *monitoringv1.ServiceMonitor) {
+	existing.Annotations = desired.Annotations
+	existing.Labels = desired.Labels
+	existing.Spec = desired.Spec
+}
+
+func mutateService(existing, desired *corev1.Service) error {
+	existing.Spec.Ports = desired.Spec.Ports
+	if err := mergeWithOverride(&existing.Spec.Selector, desired.Spec.Selector); err != nil {
+		return err
+	}
+	return nil
+}
+
+func mutateDaemonset(existing, desired *appsv1.DaemonSet) error {
+	if !existing.CreationTimestamp.IsZero() && !apiequality.Semantic.DeepEqual(desired.Spec.Selector, existing.Spec.Selector) {
```

Comment:
In general, do we not have a better way of checking if this actually exists than the creation timestamp?

Reply:
doing it this way is the standard method for operators IIRC.

Reply:
[tempo example](https://github.com/grafana/tempo-operator/blob/main/internal/manifests/mutate.go#L207-L211)